src/Analyzers/CSharp/CodeFixes/ConvertNamespace/ConvertNamespaceTransform.cs

```diff
@@ -66,51 +168,47 @@ public static BaseNamespaceDeclarationSyntax Convert(BaseNamespaceDeclarationSyn
 
         private static FileScopedNamespaceDeclarationSyntax ConvertNamespaceDeclaration(NamespaceDeclarationSyntax namespaceDeclaration)
         {
-            // We move leading and trailing trivia on the open brace to just be trailing trivia on the semicolon, so we preserve
-            // comments etc. logically at the top of the file.
-            var semiColon = SyntaxFactory.Token(SyntaxKind.SemicolonToken);
-
-            if (namespaceDeclaration.Name.GetTrailingTrivia().Any(t => t.IsSingleOrMultiLineComment()))
-            {
-                semiColon = semiColon.WithTrailingTrivia(namespaceDeclaration.Name.GetTrailingTrivia())
-                                     .WithAppendedTrailingTrivia(namespaceDeclaration.OpenBraceToken.LeadingTrivia);
-            }
-            else
-            {
-                semiColon = semiColon.WithTrailingTrivia(namespaceDeclaration.OpenBraceToken.LeadingTrivia);
-            }
-
-            semiColon = semiColon.WithAppendedTrailingTrivia(namespaceDeclaration.OpenBraceToken.TrailingTrivia);
+            var semiColon = SyntaxFactory.Token(SyntaxKind.SemicolonToken).WithoutTrivia();
 
+            // Move trivia after the original name token to now be after the new semicolon token.
             var fileScopedNamespace = SyntaxFactory.FileScopedNamespaceDeclaration(
                 namespaceDeclaration.AttributeLists,
                 namespaceDeclaration.Modifiers,
                 namespaceDeclaration.NamespaceKeyword,
                 namespaceDeclaration.Name.WithoutTrailingTrivia(),
-                semiColon,
+                semiColon.WithTrailingTrivia(namespaceDeclaration.Name.GetTrailingTrivia()),
```

Comment:
simpler logic than before. w hat was after the name should be after the semicolon.